index.js

```diff
@@ -94,6 +78,38 @@ function init(args) {
   return log(logger);
 }
 
+/**
+ * Initialize logger plugins.
+ * Accepts a comma-delimited list of logging plugins to load and
+ * composes them together to return a single function applying all plugins.
+ *
+ * @return {function}
+ */
+function initPlugins() {
+  const CLAY_LOG_PLUGINS = process.env.CLAY_LOG_PLUGINS || '';
+  const modules = CLAY_LOG_PLUGINS
+    .split(',')
+    .filter(module => !!module)
+    .filter(module => module[0] != '_') // "_" is used to reserve the private namespace.
+    .map(module => module.trim())
+    .map((module) => {
+      try {
+        return require(`./plugins/${module}`);
+      } catch (err) {
+        logger.error(`Could not locate clay-log plugin ${module}.`);
+      }
+    })
+    .filter(module => !!module);
+
+  if (modules.length == 0) {
+    return (x) => x;
+  } else if (modules.length == 1) {
+    return modules[0];
+  } else {
+    return modules.reduceRight((a, b) => (...args) => b(a(...args)));
```

Comment:
I'm not exactly sure what this is doing. why not just return a function that maps over each module and requires it?

Reply:
Let's say we have `CLAY_LOG_PLUGINS=a,b,c`.

Each function accepts an argument `x` which represents an instance of `pino`'s logger.

This bit of code is composing a new function `c(b(a(x)))` rather than doing this:

```
c(x)
b(x)
a(x)
```